pontoon/base/forms.py

```diff
@@ -158,9 +128,29 @@ def save(self, *args, **kwargs):
         translators = self.cleaned_data.get("translators", User.objects.none())
         managers = self.cleaned_data.get("managers", User.objects.none())
 
+        before_count = self.user.badges_promotion_count
+
         self.assign_users_to_groups("translators", translators)
         self.assign_users_to_groups("managers", managers)
 
+        after_count = self.user.badges_promotion_count
+
+        # Award Community Builder badge
+        if (
+            after_count > before_count
+            and after_count in settings.BADGES_PROMOTION_THRESHOLDS
+        ):
+            self.community_builder_level_reached = (
+                settings.BADGES_PROMOTION_THRESHOLDS.index(after_count) + 1
+            )
+            _send_badge_notification(
+                self.user,
+                "Community Builder Badge",
+                self.community_builder_level_reached,
+            )
+
+        return self.community_builder_level_reached
+
```

Comment:
Will this code work now if you move it back to `assign_users_to_groups()`? 

That way promotions of users in per-project permissions would also count against badges. The spec isn't clear if this is a requirement, but it would be nice to check if the current code supports it.